File: EventProcessor/Node.js/Web/Plain/src/queue/QueueAdapter.js
class QueueAdapter {
  async connect() { throw new Error('Not implemented'); }
  async publish(queue, message) { throw new Error('Not implemented'); }
  async subscribe(queue, handler, options = {}) { throw new Error('Not implemented'); }
  async ack(message) { throw new Error('Not implemented'); }
  async nack(message, requeue = false) { throw new Error('Not implemented'); }
  async close() { throw new Error('Not implemented'); }
}
module.exports = QueueAdapter;

File: EventProcessor/Java/Web/SpringBoot/src/main/java/com/portfolio/eventprocessor/queue/QueueAdapter.java
package com.portfolio.eventprocessor.queue;

import java.util.Map;
import java.util.function.BiConsumer;

public interface QueueAdapter {
    void connect();
    void publish(String queue, Map<String, Object> message);
    void subscribe(String queue, BiConsumer<Map<String, Object>, String> handler);
    void ack(String messageId);
    void nack(String messageId, boolean requeue);
    void close();
}
